modules/renter.go

```diff
@@ -55,19 +55,18 @@ type ContractUtility struct {
 // DownloadInfo provides information about a file that has been requested for
 // download.
 type DownloadInfo struct {
-	SiaPath     string         `json:"siapath"`
-	Destination DownloadWriter `json:"destination"`
-	Filesize    uint64         `json:"filesize"`
-	Received    uint64         `json:"received"`
-	StartTime   time.Time      `json:"starttime"`
-	Error       string         `json:"error"`
-}
-
-// DownloadWriter provides an interface which all output writers have to implement.
-type DownloadWriter interface {
-	WriteAt(b []byte, off int64) (int, error)
-	Destination() string
-	Close() error
+	Destination     string `json:"destination"`     // The destination of the download.
+	DestinationType string `json:"destinationtype"` // Can be "file", "memory buffer", or "http stream".
+	Length          uint64 `json:"length"`          // The length requested for the download.
+	Offset          uint64 `json:"offset"`          // The offset within the siafile requested for the download.
+	SiaPath         string `json:"siapath"`         // The siapath of the file used for the download.
+
+	Completed           bool      `json:"completed"`           // Whether or not the download has completed.
```

Comment:
Is this ever set? I can't find it. If it is, we should probably make sure it is set after we wrote the file to the writer to make sure it is actually on disk if the flag is set to true.

Reply:
download.go, line 475

We don't set it until after the file has been fully written to the writer.

Reply:
or rather, I should say that it won't be set to true until after the file has been fully written to the writer, and the writer closed.